xcm-simulator/src/lib.rs

```diff
@@ -589,6 +589,36 @@ mod tests {
 		});
 	}
 
+	#[test]
+	fn event_collection_works() {
+		init_tracing();
+
+		MockNet::reset();
+
+		const AMOUNT: u128 = trappist::EXISTENTIAL_DEPOSIT * 10;
+		const MAX_WEIGHT: u128 = 1_000_000_000;
+
+		Trappist::execute_with(|| {
+			assert_ok!(trappist::PolkadotXcm::execute(
```

Comment:
If the goal is simply to emit an event, it's not clear why we're actually doing a `WithdrawAsset` XCM here.

Something like this makes the intention more explicit:
```
assert_ok!(trappist::System::remark_with_event(trappist::RuntimeOrigin::signed(ALICE), vec![1]));
```

Reply:
The only reason for these is that they actually trigger multiple events, most importantly from the balances pallet when fees are charged, which was the issue you encountered and this PR addresses. 

Perhaps a simple rename of the test to something like `execute_deposits_events`?

Reply:
ok, I was missing that point, that makes sense then.